spec/AMQP/20-AMQP_FORMAT.md

```diff
@@ -1,3 +1,118 @@
 # AMQP format
 
-TBD
\ No newline at end of file
+The Advanced Message Queuing Protocol (AMQP) is an open internet protocol for
+business messaging. It defines a binary wire-level protocol that allows for the
+reliable exchange of business messages between two parties. AMQP can be used as
+a protocol for asynchronous communication between components of an application.
+From the distributed tracing and telemetry correlation perspective it is a
+known problem to be able to correlate a component that placed message and
+component that processed it later. This specification describes how trace
+context MUST be encoded into AMQP messages.
+
+## Trace context fields placement in a message
+
+AMQP defines message as a payload with the additional annotations sections.
+There are two annotation sections this specification refers -
+"application-properties" and "message-annotations". See
+[3.2](http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-messaging-v1.0-os.html#section-message-format)
+of OASIS Advanced Message Queuing Protocol (AMQP) Version 1.0, Part 3:
+Messaging.
+
+AMQP message section "application-properties" is immutable collection of
+properties that is defined by message publisher and can be read by the message
+consumer. Message brokers cannot mutate those properties. Section
+"message-annotations" is designed for message brokers to use and can be mutated
+during the message processing.
+
+Fields `traceparent` and `tracestate` SHOULD be added to the message in the
+application-properties section by message publisher. Once the message has been
+created, it’s no longer permissible to edit the bare message. So if it were
+necessary to annotate the message inside the middleware as it transits, that
+MUST happen in the “message-annotations” section, using the
```

Comment:
so the trace context MAY be added to the message-annotations or application-properties depending on the way it was added. Assuming message-annotations could be accessed and changed at any moment,  why do we need to use application-properties at all?
it creates a bit of extra complexity and perf impact.

Reply:
I've added rationale section for this. My understanding that it is desire-able and easier to use app properties. So message annotations is simply a fallback option.

Reply:
@SergeyKanzhelev - I am not quite convinced by that rationale. Having to look for trace context in two different locations and support some sort of overlay logic seems overly complex & unnecessary. What benefits does it provide?

Reply:
My understanding is that applicaiton properties is easier to implement and would work fine in all cases, except when broker need to change this context and they cannot change applicaiton properties according to spec.

Reply:
I've unresolved conversation to keep feedback coming